test/stage-options.test.ts

```diff
@@ -196,4 +197,50 @@ describe('job settings', () => {
       expect(readFileSync(pipeline.workflowPath, 'utf-8')).toContain('if: github.repository == \'github/repo\'\n');
     });
   });
+
+  test('can set pre/post github action job step', () => {
+    withTemporaryDirectory((dir) => {
+      const pipeline = new GitHubWorkflow(app, 'Pipeline', {
+        workflowPath: `${dir}/.github/workflows/deploy.yml`,
+        synth: new ShellStep('Build', {
+          installCommands: ['yarn'],
+          commands: ['yarn build'],
+        }),
+      });
+
+      const stage = new Stage(app, 'MyStack', {
+        env: { account: '111111111111', region: 'us-east-1' },
+      });
+
+      new Stack(stage, 'MyStack');
+
+      pipeline.addStageWithGitHubOptions(stage, {
+        pre: [new GithubActionStep('PreDeployAction', {
+          jobStep: {
+            name: 'pre deploy action',
+            uses: 'my-pre-deploy-action@1.0.0',
+            with: {
+              'app-id': 1234,
+              'secrets': 'my-secrets',
+            },
+          },
+        })],
+        post: [new GithubActionStep('PostDeployAction', {
+          jobStep: {
+            name: 'post deploy action',
+            uses: 'my-post-deploy-action@1.0.0',
+            with: {
+              'app-id': 4321,
+              'secrets': 'secrets',
+            },
+          },
+        })],
+      });
+
+      app.synth();
+
+      expect(readFileSync(pipeline.workflowPath, 'utf-8')).toContain('my-pre-deploy-action\@1\.0\.0');
+      expect(readFileSync(pipeline.workflowPath, 'utf-8')).toContain('my-post-deploy-action\@1\.0\.0');
```

Comment:
I know I've said otherwise before, but I think this change deserves a snapshot test in addition (can be the same test).

Reply:
added snapshot test